File: KalafcheBackend/src/main/java/com/kalafche/service/DailyStoreReportService.java
package com.kalafche.service;

import java.util.List;

import com.kalafche.exceptions.CommonException;
import com.kalafche.model.DailyStoreReport;
import com.kalafche.model.DayInMillis;

public interface DailyStoreReportService {

	DailyStoreReport finalizeDailyStoreReport(Integer storeId);

	DailyStoreReport getDailyStoreReportById(Integer dailyStoreReportId);

	DailyStoreReport getDailyStoreReportByDay(Integer storeId, DayInMillis day);

	DailyStoreReport calculateDailyStoreReport(Integer storeId, Boolean scheduled);

	List<DailyStoreReport> searchDailyStoreReports(Long startDateMilliseconds, Long endDateMilliseconds,
			String storeIds);

	Boolean canDailyStoreReportBeFinalized(Integer storeId, Boolean scheduled);

	List<DailyStoreReport> searchDailyCompanyReports(Long startDateMilliseconds, Long endDateMilliseconds, Integer companyId);

	void updateDailyStoreReport(DailyStoreReport dailyStoreReport) throws CommonException;

	List<DailyStoreReport> searchDailyStoreReportsForCorrection(Long startDateMilliseconds, Long endDateMilliseconds, String storeIds) throws CommonException;

}

File: KalafcheFrontend/app/scripts/services/DailyStoreReportService.js
'use strict';

angular.module('kalafcheFrontendApp')
	.service('DailyStoreReportService', function($http, Environment) {
		angular.extend(this, {
			searchDailyStoreReports: searchDailyStoreReports,
            searchDailyStoreReportsForCorrection: searchDailyStoreReportsForCorrection,
            searchDailyCompanyReports: searchDailyCompanyReports,
            calculateDailyStoreReport: calculateDailyStoreReport,
            finalizeDailyStoreReport: finalizeDailyStoreReport,
            isDailyStoreReportCanBeFinalized: isDailyStoreReportCanBeFinalized,
            updateDailyStoreReport: updateDailyStoreReport
		});

        function searchDailyStoreReports(startDateMilliseconds, endDateMilliseconds, storeIds) { 
            var params = {"params" : 
                {"startDateMilliseconds": startDateMilliseconds, 
                "endDateMilliseconds": endDateMilliseconds, 
                "storeIds": storeIds}};
            console.log(params);

            return $http.get(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport', params)
                .then(
                    function(response) {
                        console.log(response.data);
                        return response.data
                    }
                );
        }

        function searchDailyStoreReportsForCorrection() { 
            return $http.get(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport/correction')
                .then(
                    function(response) {
                        console.log(response.data);
                        return response.data
                    }
                );
        }

        function searchDailyCompanyReports(startDateMilliseconds, endDateMilliseconds, companyId) { 
            var params = {"params" : 
                {"startDateMilliseconds": startDateMilliseconds, 
                "endDateMilliseconds": endDateMilliseconds,
                "companyId": companyId}};
            console.log(params);

            return $http.get(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport/companies', params)
                .then(
                    function(response) {
                        console.log(response.data);
                        return response.data
                    }
                );
        }

        function calculateDailyStoreReport(storeId) {
            return $http.get(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport/current/' + storeId)
                .then(
                    function(response) {
                        return response.data
                    }
                );
        }

        function finalizeDailyStoreReport(storeId) {
            return $http.put(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport/' + storeId)
                    .then(
                        function(response) {
                            return response.data;
                        }
                    )
        }

        function isDailyStoreReportCanBeFinalized(storeId) {
            return $http.get(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport/checkFinalization/' + storeId)
                    .then(
                        function(response) {
                            return response.data;
                        }
                    )
        }

        function updateDailyStoreReport(dailyStoreReport) { 
            return $http.post(Environment.apiEndpoint + '/KalafcheBackend/dailyStoreReport', dailyStoreReport)
                .then(
                    function(response) {
                        return response.data;
                    }
                )
        }

	});